benchmark/README.md

```diff
@@ -1,417 +1,246 @@
+        <code>url</code> implementation and the WHATWG URL implementation.
+      </td>
+    </tr>
+    <tr>
+      <td>util</td>
+      <td>
+        Benchmarks for the <code>util</code> subsystem.
+      </td>
+    </tr>
+    <tr>
+      <td>vm</td>
+      <td>
+        Benchmarks for the <code>vm</code> subsystem.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+### Other Top-level files
+
+The top-level files include common dependencies of the benchmarks
+and the tools for launching benchmarks and visualizing their output.
+The actual benchmark scripts should be placed in their corresponding
+directories.
+
+* `_benchmark_progress.js`: implements the progress bar displayed
+  when running `compare.js`
+* `_cli.js`: parses the command line arguments passed to `compare.js`,
+  `run.js` and `scatter.js`
+* `_cli.R`: parses the command line arguments passed to `compare.R`
+* `_http-benchmarkers.js`: selects and runs external tools for benchmarking
+  the `http` subsystem.
+* `common.js`: see [Common API](#common-api).
+* `compare.js`: command line tool for comparing performance between different
+  Node.js binaries.
+* `compare.R`: R script for statistically analyzing the output of
+  `compare.js`
+* `run.js`: command line tool for running individual benchmark suite(s).
+* `scatter.js`: command line tool for comparing the performance
+  between different parameters in benchmark configurations,
+  for example to analyze the time complexity.
+* `scatter.R`: R script for visualizing the output of `scatter.js` with
+  scatter plots.
+
+## Common API
+
+The common.js module is used by benchmarks for consistency across repeated
+tasks. It has a number of helpful functions and properties to help with
+writing benchmarks.
+
+### createBenchmark(fn, configs[, options])
+
+See [the guide on writing benchmarks](../doc/guides/writing-and-running-benchmarks.md#basics-of-a-benchmark).
+
+### default\_http\_benchmarker
+
+The default benchmarker used to run HTTP benchmarks.
+See [the guide on writing HTTP benchmarks](../doc/guides/writing-and-running-benchmarks.md#creating-an-http-benchmark).
+
+
+### PORT
+
+The default port used to run HTTP benchmarks.
+See [the guide on writing HTTP benchmarks](../doc/guides/writing-and-running-benchmarks.md#creating-an-http-benchmark).
+
+### sendResult(data)
+
+Used in special benchmarks that can't use `createBenchmark` and the object
+it returns to accomplish what they need. This function reports timing
+data to the parent process (usually created by running `compare.js`, `run.js` or
+`scatter.js`).
+
+### v8ForceOptimization(method)
+
+Force V8 to optimize the function with the native function
+`%OptimizeFunctionOnNextCall()` and return the optimization status
+after that.
+
+It is used to prevent the benchmark from getting disrupted
+by the optimizer kicking in halfway through. However, this could result in a
+less effective optimization, so it should be used with caution.
```

Comment:
I think the reasons for a less effective optimization are probably too many to explain here(no proper inlining because the profiler doesn't know the functions calls inside are hot, no proper type feedback, possible deopt, etc)..probably better left for another PR to update the guide instead of a readme.